sdk/rust/oak/src/invocations.rs

```diff
@@ -0,0 +1,156 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::io::{ReceiverExt, SenderExt};
+use log::warn;
+
+/// An invocation, consisting of exactly two channels: one to read incoming requests from the
+/// client, and one to write outgoing responses to the client.
+pub struct Invocation<
+    R: prost::Message + crate::handle::HandleVisit + crate::io::Decodable + std::default::Default,
+    S: prost::Message + crate::handle::HandleVisit + crate::io::Encodable + std::clone::Clone,
```

Comment:
Do we really need all these type bounds? I would have expected just `R: Decodable, S: Encodable`.